src/main/java/org/craftercms/studio/controller/rest/v2/ConfigurationController.java

```diff
@@ -85,13 +86,19 @@ public ResponseBody clearCache(@EsapiValidatedParam(type = SITE_ID) @RequestPara
     public ResponseBody getConfiguration(@EsapiValidatedParam(type = SITE_ID) @RequestParam(name = "siteId", required = true) String siteId,
                                          @EsapiValidatedParam(type = ALPHANUMERIC) @RequestParam(name = "module", required = true) String module,
                                          @EsapiValidatedParam(type = HTTPURI) @RequestParam(name = "path", required = true) String path,
-                                         @EsapiValidatedParam(type = ALPHANUMERIC, notNull = false, notEmpty = false, notBlank = false) @RequestParam(name = "environment", required = false) String environment) {
+                                         @EsapiValidatedParam(type = ALPHANUMERIC, notNull = false, notEmpty = false, notBlank = false) @RequestParam(name = "environment", required = false) String environment)
+            throws ContentNotFoundException {
         final String content;
         if (StringUtils.equals(siteId, studioConfiguration.getProperty(CONFIGURATION_GLOBAL_SYSTEM_SITE))) {
             content = configurationService.getGlobalConfigurationAsString(path);
         } else {
             content = configurationService.getConfigurationAsString(siteId, module, path, environment);
         }
+
+        if (content == null) {
```

Comment:
Move the check and exception throwing to the service layer.

Reply:
Updated to throw exception for method `getConfigurationAsString`